src/functions.h

```diff
@@ -935,6 +935,33 @@ class ACoth: public HyperbolicFunction {
 //! Canonicalize ACoth:
 RCP<const Basic> acoth(const RCP<const Basic> &arg);
 
+class ASech: public HyperbolicFunction {
+//! The inverse hyperbolic secant function.
+public:
+    IMPLEMENT_TYPEID(ASECH)
+    //! ASech Constructor
+    ASech(const RCP<const Basic> &arg);
+    /*! Equality comparator
+     * \param o - Object to be compared with
+     * \return whether the 2 objects are equal
+     * */
+    virtual bool __eq__(const Basic &o) const;
+    virtual int compare(const Basic &o) const;
+    //! \return stringify version
+    virtual std::string __str__() const;
+    //! \return `true` if canonical
```

Comment:
Now when #423 was merged, the `__str__()` method should be removed and rather implemented in the `StrPrinter`.